<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>3&nbsp; Results – Msc Bioinformatics thesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/04-Discussion.html" rel="next">
<link href="../chapters/02-Materials-and-methods.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-fed3a45c7a334f7a8f106de68cf0fca8.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-aa42bd5397b87ae3001db572b235c885.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<link href="https://fonts.googleapis.com/css?family=Fira+Sans|Merriweather|Source Code Pro" rel="stylesheet">


</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/03-Results.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Results</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Msc Bioinformatics thesis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/vgoupille/Rapport_stage" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="../thesis.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Front matter</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/01-Introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/02-Materials-and-methods.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Materials and Methods</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/03-Results.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Results</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/04-Discussion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Discussion</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/05-Conclusion-and-future-work.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Conclusion and Future Work</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/06-References.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bibliography</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/A-Appendix.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Annexe A </span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/B-Appendix.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">B</span>&nbsp; <span class="chapter-title">Annexe B</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/C-Appendix.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">C</span>&nbsp; <span class="chapter-title">Annexe C</span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title"><b>Sections</b></h2>
   
  <ul class="collapse">
  <li><a href="#stats-sur-les-reads-r1-et-r2" id="toc-stats-sur-les-reads-r1-et-r2" class="nav-link active" data-scroll-target="#stats-sur-les-reads-r1-et-r2"><span class="header-section-number">3.1</span> Stats sur les reads R1 et R2 :</a></li>
  <li><a href="#trimming" id="toc-trimming" class="nav-link" data-scroll-target="#trimming"><span class="header-section-number">3.2</span> Trimming</a></li>
  <li><a href="#starsolo" id="toc-starsolo" class="nav-link" data-scroll-target="#starsolo"><span class="header-section-number">3.3</span> STARsolo</a></li>
  <li><a href="#genome" id="toc-genome" class="nav-link" data-scroll-target="#genome"><span class="header-section-number">3.4</span> Genome</a></li>
  <li><a href="#transcriptome" id="toc-transcriptome" class="nav-link" data-scroll-target="#transcriptome"><span class="header-section-number">3.5</span> Transcriptome</a></li>
  <li><a href="#fitrage-des-cellules" id="toc-fitrage-des-cellules" class="nav-link" data-scroll-target="#fitrage-des-cellules"><span class="header-section-number">4</span> Fitrage des cellules</a></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section"><span class="header-section-number">5</span> </a>
  <ul class="collapse">
  <li><a href="#summary-of-results" id="toc-summary-of-results" class="nav-link" data-scroll-target="#summary-of-results"><span class="header-section-number">5.1</span> Summary of results</a></li>
  <li><a href="#multiqc-quality-reports" id="toc-multiqc-quality-reports" class="nav-link" data-scroll-target="#multiqc-quality-reports"><span class="header-section-number">5.2</span> MultiQC Quality Reports</a></li>
  <li><a href="#genfull" id="toc-genfull" class="nav-link" data-scroll-target="#genfull"><span class="header-section-number">5.3</span> Genfull</a></li>
  <li><a href="#genefull-summary-stats" id="toc-genefull-summary-stats" class="nav-link" data-scroll-target="#genefull-summary-stats"><span class="header-section-number">5.4</span> Genefull summary stats</a></li>
  <li><a href="#interpretation-of-starsolo-results" id="toc-interpretation-of-starsolo-results" class="nav-link" data-scroll-target="#interpretation-of-starsolo-results"><span class="header-section-number">5.5</span> Interpretation of STARsolo Results</a></li>
  <li><a href="#apres-starsolo-mettre-le-nombre-de-reads-avec-valid-barcodes-dans-la-table" id="toc-apres-starsolo-mettre-le-nombre-de-reads-avec-valid-barcodes-dans-la-table" class="nav-link" data-scroll-target="#apres-starsolo-mettre-le-nombre-de-reads-avec-valid-barcodes-dans-la-table"><span class="header-section-number">5.6</span> apres starsolo mettre le nombre de reads avec valid barcodes dans la table</a></li>
  </ul></li>
  <li><a href="#initialize-the-seurat-object-with-the-raw-non-normalized-data." id="toc-initialize-the-seurat-object-with-the-raw-non-normalized-data." class="nav-link" data-scroll-target="#initialize-the-seurat-object-with-the-raw-non-normalized-data."><span class="header-section-number">6</span> Initialize the Seurat object with the raw (non-normalized data).</a>
  <ul class="collapse">
  <li><a href="#overview" id="toc-overview" class="nav-link" data-scroll-target="#overview"><span class="header-section-number">6.1</span> Overview</a></li>
  <li><a href="#single-cell-rna-seq-analysis" id="toc-single-cell-rna-seq-analysis" class="nav-link" data-scroll-target="#single-cell-rna-seq-analysis"><span class="header-section-number">6.2</span> Single-cell RNA-seq Analysis</a></li>
  <li><a href="#functional-analysis" id="toc-functional-analysis" class="nav-link" data-scroll-target="#functional-analysis"><span class="header-section-number">6.3</span> Functional Analysis</a></li>
  <li><a href="#integration-with-previous-studies" id="toc-integration-with-previous-studies" class="nav-link" data-scroll-target="#integration-with-previous-studies"><span class="header-section-number">6.4</span> Integration with Previous Studies</a></li>
  <li><a href="#summary-of-key-findings" id="toc-summary-of-key-findings" class="nav-link" data-scroll-target="#summary-of-key-findings"><span class="header-section-number">6.5</span> Summary of Key Findings</a></li>
  </ul></li>
  </ul>
<div class="toc-actions"><ul class="collapse"><li><a href="https://github.com/vgoupille/Rapport_stage/edit/main/chapters/03-Results.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div><div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="https://data.nasa.gov/"><i class="bi bi-link-45deg"></i>NASA Open Data</a></li></ul></div><div class="quarto-code-links"><h2>Code Links</h2><ul><li><a href="https://github.com/vgoupille/Analyse_data/blob/d77d4ff43dd2435951e23d63cc3e38a009653760/analyse/5_Filter_Seurat_Object/1_Filter-GeneType_SeuratObject.R"><i class="bi bi-file-code"></i>Data Import Code</a></li><li><a href="../data/test.py"><i class="bi bi-file-code"></i>script test</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Results</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<ul>
<li>4 libraries of unequal sizes (see Solène’s explanation for why they were not exactly equivalent)</li>
<li>This impacts library efficiency</li>
<li>=&gt; Recommendation: balance the libraries for optimal results</li>
</ul>
<section id="stats-sur-les-reads-r1-et-r2" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="stats-sur-les-reads-r1-et-r2"><span class="header-section-number">3.1</span> Stats sur les reads R1 et R2 :</h2>
<ul>
<li>On a subset of 1,000,000 reads:
<ul>
<li>Percentage of reads containing TSO</li>
<li>Percentage of reads containing polyA</li>
<li>Percentage of reads containing adapter</li>
<li>Percentage of reads containing linker -&gt; possibly in appendix</li>
</ul></li>
</ul>
<p>Reminder about saturation calculation method:</p>
</section>
<section id="trimming" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="trimming"><span class="header-section-number">3.2</span> Trimming</h2>
<p>ici j’ai suivi les recommendations de kuchina , seulement presenté les resultats Genfull , mettre plus tard les resultats de Gene</p>
<ul>
<li>renvoie vers l’annexe pour les multiqc (fastp, cutadapt) avant et apres trimming</li>
<li></li>
<li>mais au final on obtient des resultats tres interessants et propres</li>
</ul>
<p>=&gt; peut etre il aurait été interressant de faire un trimming comme kuchina 2021 pour comparer les resultats =&gt; renvoie vers la discussion pour le trimming fait dans l’article de<span class="citation" data-cites="kuchina2021"><sup><a href="06-References.html#ref-kuchina2021" role="doc-biblioref">1</a></sup></span> - avait presque 90% de saturation =&gt; a verifier si c’est vraiment le cas - et 10 % des sequences avec TSO (voir Annexe)</p>
<ul>
<li>Summary table of Starsolo results</li>
<li>numbers of reads before and after trimming</li>
<li>taux de saturation …</li>
<li>=&gt; renvoie vers la discussion pour les taux de saturation <!-- 
// figure - ![](../figures/trimming_quality.png) --></li>
</ul>
</section>
<section id="starsolo" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="starsolo"><span class="header-section-number">3.3</span> STARsolo</h2>
<p>-pour starsolo renvoie un gene qui serait mal annoté donc est automatiquement ignoré</p>
<p>commande starsolo suivante comme dans bretn permet UMI a une erreur… possition des barcodes</p>
</section>
<section id="genome" class="level2" data-number="3.4">
<h2 data-number="3.4" class="anchored" data-anchor-id="genome"><span class="header-section-number">3.4</span> Genome</h2>
</section>
<section id="transcriptome" class="level2" data-number="3.5">
<h2 data-number="3.5" class="anchored" data-anchor-id="transcriptome"><span class="header-section-number">3.5</span> Transcriptome</h2>
<!-- -   figure de % de type of RNA
-   ![](../figures/pourcentage_RNA.png) -->
</section>
<section id="fitrage-des-cellules" class="level1" data-number="4">
<h1 data-number="4"><span class="header-section-number">4</span> Fitrage des cellules</h1>
<ul>
<li><p>reflexion sur le filtrages des cellules est complexe : comme mentionnée dans l’article de il existe des methodes de filtrages plus ou moins complexes<br>
-filtrage sur le types de reads, globales ou seuils differents entre les differentes conditions biologiques -filtrage sur le nombre de reads par cellule -filtrage sur le nombre de genes exprimés par cellule -filtrage sur le nombre de reads par gene</p></li>
<li><p>filtrage avec un threashold</p></li>
<li><p>dans l’article de kuchina 2021 , ils ont utilisé un seuil de 200 UMI par cellule</p></li>
<li></li>
<li><p>preprint de … pourrait etre interessant de se focus aussi sur rRNA (lien avec growth rate)</p></li>
</ul>
</section>
<section id="section" class="level1" data-number="5">
<h1 data-number="5"><span class="header-section-number">5</span> </h1>
<p>housekeeping genes</p>
<p>choix ou non de pooler les replicas techniques ensembles</p>
<section id="summary-of-results" class="level2" data-number="5.1">
<h2 data-number="5.1" class="anchored" data-anchor-id="summary-of-results"><span class="header-section-number">5.1</span> Summary of results</h2>
</section>
<section id="multiqc-quality-reports" class="level2" data-number="5.2">
<h2 data-number="5.2" class="anchored" data-anchor-id="multiqc-quality-reports"><span class="header-section-number">5.2</span> MultiQC Quality Reports</h2>
<p>Detailed sequence quality reports are available below. Click on each report to view it.</p>
<div class="tabset-margin-container"></div><div class="panel-tabset">
<ul class="nav nav-tabs" role="tablist"><li class="nav-item" role="presentation"><a class="nav-link active" id="tabset-1-1-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-1" role="tab" aria-controls="tabset-1-1" aria-selected="true" aria-current="page">MultiQC Report R1 Before Trimming</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-2-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-2" role="tab" aria-controls="tabset-1-2" aria-selected="false">MultiQC Report R1 After Trimming</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-3-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-3" role="tab" aria-controls="tabset-1-3" aria-selected="false">MultiQC Report R2 Before Trimming</a></li><li class="nav-item" role="presentation"><a class="nav-link" id="tabset-1-4-tab" data-bs-toggle="tab" data-bs-target="#tabset-1-4" role="tab" aria-controls="tabset-1-4" aria-selected="false">MultiQC Report R2 After Trimming</a></li></ul>
<div class="tab-content">
<div id="tabset-1-1" class="tab-pane active" role="tabpanel" aria-labelledby="tabset-1-1-tab">
<details>
<summary>
Click to view MultiQC report R1 before trimming
</summary>
<iframe src="../docs/multiqc_reports/multiqc_report_R1_before_trimming.html" width="100%" height="800px" frameborder="0" loading="lazy">
</iframe>
</details>
</div>
<div id="tabset-1-2" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-2-tab">
<details>
<summary>
Click to view MultiQC report R1 after trimming
</summary>
<iframe src="../docs/multiqc_reports/multiqc_report_R1_after_trimming.html" width="100%" height="800px" frameborder="0" loading="lazy">
</iframe>
</details>
</div>
<div id="tabset-1-3" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-3-tab">
<details>
<summary>
Click to view MultiQC report R2 before trimming
</summary>
<iframe src="../docs/multiqc_reports/multiqc_report_R2_before_trimming.html" width="100%" height="800px" frameborder="0" loading="lazy">
</iframe>
</details>
</div>
<div id="tabset-1-4" class="tab-pane" role="tabpanel" aria-labelledby="tabset-1-4-tab">
<details>
<summary>
Click to view MultiQC report R2 after trimming
</summary>
<iframe src="../docs/multiqc_reports/multiqc_report_R2_after_trimming.html" width="100%" height="800px" frameborder="0" loading="lazy">
</iframe>
</details>
</div>
</div>
</div>
<p>Key quality metrics are summarized in the tables below.</p>
<div id="tbl-example" class="quarto-layout-panel anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5.1: Before trimming
</figcaption>
<div aria-describedby="tbl-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;">Sample Name</th>
<th style="text-align: left;">Dups</th>
<th style="text-align: left;">GC</th>
<th style="text-align: left;">Median len</th>
<th style="text-align: left;">Seqs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BC_0076_R1</td>
<td style="text-align: left;">94.5%</td>
<td style="text-align: left;">55.0%</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">631.4M</td>
</tr>
<tr class="even">
<td style="text-align: left;">BC_0077_R1</td>
<td style="text-align: left;">94.1%</td>
<td style="text-align: left;">53.0%</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">325.5M</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BC_0079_R1</td>
<td style="text-align: left;">93.8%</td>
<td style="text-align: left;">53.0%</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">379.1M</td>
</tr>
<tr class="even">
<td style="text-align: left;">BC_0080_R1</td>
<td style="text-align: left;">94.6%</td>
<td style="text-align: left;">54.0%</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">397.7M</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong>Total</strong></td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;"><strong>1,733.7M</strong></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</figure>
</div>
<div id="tbl-example" class="quarto-layout-panel anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5.2: After trimming
</figcaption>
<div aria-describedby="tbl-example-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;">Sample Name</th>
<th style="text-align: left;">Dups</th>
<th style="text-align: left;">GC</th>
<th style="text-align: left;">Median len</th>
<th style="text-align: left;">Seqs</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">BC_0076_R1</td>
<td style="text-align: left;">98.7%</td>
<td style="text-align: left;">51.0%</td>
<td style="text-align: left;">127bp</td>
<td style="text-align: left;">450.8M</td>
</tr>
<tr class="even">
<td style="text-align: left;">BC_0077_R1</td>
<td style="text-align: left;">98.6%</td>
<td style="text-align: left;">51.0%</td>
<td style="text-align: left;">157bp</td>
<td style="text-align: left;">248.4M</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BC_0079_R1</td>
<td style="text-align: left;">98.6%</td>
<td style="text-align: left;">51.0%</td>
<td style="text-align: left;">152bp</td>
<td style="text-align: left;">285.2M</td>
</tr>
<tr class="even">
<td style="text-align: left;">BC_0080_R1</td>
<td style="text-align: left;">98.7%</td>
<td style="text-align: left;">51.0%</td>
<td style="text-align: left;">132bp</td>
<td style="text-align: left;">300.1M</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong>Total</strong></td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
<td style="text-align: left;"><strong>1,284.5M</strong></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</figure>
</div>
<div id="tbl-summary" class="quarto-layout-panel anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5.3: Summary of sequence metrics before and after trimming, including percentage changes
</figcaption>
<div aria-describedby="tbl-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<table class="caption-top table">
<colgroup>
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
<col style="width: 14%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Sample Name</th>
<th style="text-align: left;">Before Trimming</th>
<th style="text-align: left;"></th>
<th style="text-align: left;">After Trimming</th>
<th style="text-align: left;"></th>
<th style="text-align: left;">Change in</th>
<th style="text-align: left;"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"></td>
<td style="text-align: left;">Median len</td>
<td style="text-align: left;">Seqs</td>
<td style="text-align: left;">Median len</td>
<td style="text-align: left;">Seqs</td>
<td style="text-align: left;">Length</td>
<td style="text-align: left;">Sequences</td>
</tr>
<tr class="even">
<td style="text-align: left;">————–</td>
<td style="text-align: left;">—————-</td>
<td style="text-align: left;">————</td>
<td style="text-align: left;">—————-</td>
<td style="text-align: left;">————</td>
<td style="text-align: left;">———–</td>
<td style="text-align: left;">———–</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BC_0076_R1</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">631.4M</td>
<td style="text-align: left;">127bp</td>
<td style="text-align: left;">450.8M</td>
<td style="text-align: left;">-47.3%</td>
<td style="text-align: left;">-28.6%</td>
</tr>
<tr class="even">
<td style="text-align: left;">BC_0077_R1</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">325.5M</td>
<td style="text-align: left;">157bp</td>
<td style="text-align: left;">248.4M</td>
<td style="text-align: left;">-34.9%</td>
<td style="text-align: left;">-23.7%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">BC_0079_R1</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">379.1M</td>
<td style="text-align: left;">152bp</td>
<td style="text-align: left;">285.2M</td>
<td style="text-align: left;">-36.9%</td>
<td style="text-align: left;">-24.8%</td>
</tr>
<tr class="even">
<td style="text-align: left;">BC_0080_R1</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">397.7M</td>
<td style="text-align: left;">132bp</td>
<td style="text-align: left;">300.1M</td>
<td style="text-align: left;">-45.2%</td>
<td style="text-align: left;">-24.5%</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong>Mean</strong></td>
<td style="text-align: left;"><strong>241bp</strong></td>
<td style="text-align: left;"><strong>433.4M</strong></td>
<td style="text-align: left;"><strong>142bp</strong></td>
<td style="text-align: left;"><strong>321.1M</strong></td>
<td style="text-align: left;">-</td>
<td style="text-align: left;">-</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>Total</strong></td>
<td style="text-align: left;">-</td>
<td style="text-align: left;"><strong>1733.7M</strong></td>
<td style="text-align: left;">-</td>
<td style="text-align: left;"><strong>1284.5M</strong></td>
<td style="text-align: left;"><strong>-41.1%</strong></td>
<td style="text-align: left;"><strong>-25.4%</strong></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</figure>
</div>
<div id="tbl-starsolo" class="quarto-layout-panel anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-starsolo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5.4: STARsolo barcode statistics
</figcaption>
<div aria-describedby="tbl-starsolo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;">Metric</th>
<th style="text-align: left;">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">nNoAdapter</td>
<td style="text-align: left;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">nNoUMI</td>
<td style="text-align: left;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nNoCB</td>
<td style="text-align: left;">67,177</td>
</tr>
<tr class="even">
<td style="text-align: left;">nNinCB</td>
<td style="text-align: left;">0</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nNinUMI</td>
<td style="text-align: left;">16,893,550</td>
</tr>
<tr class="even">
<td style="text-align: left;">nUMIhomopolymer</td>
<td style="text-align: left;">1,697,129</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nTooMany</td>
<td style="text-align: left;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">nNoMatch</td>
<td style="text-align: left;">163,166,830</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nMismatchesInMultCB</td>
<td style="text-align: left;">3,323,192</td>
</tr>
<tr class="even">
<td style="text-align: left;">nExactMatch</td>
<td style="text-align: left;">1,046,121,284</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nMismatchOneWL</td>
<td style="text-align: left;">53,206,471</td>
</tr>
<tr class="even">
<td style="text-align: left;">nMismatchToMultWL</td>
<td style="text-align: left;">0</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</figure>
</div>
<section id="barcode-statistics-interpretation" class="level3">
<h3 class="anchored" data-anchor-id="barcode-statistics-interpretation">Barcode Statistics Interpretation</h3>
<p>The analysis of cell barcodes reveals several important points about our data quality:</p>
<ul>
<li><strong>Barcode Quality</strong>:
<ul>
<li>The absence of reads without adapter (<code>nNoAdapter = 0</code>) and without UMI (<code>nNoUMI = 0</code>) indicates excellent library preparation quality</li>
<li>The relatively low number of reads without cell barcode (<code>nNoCB = 67,177</code>) represents less than 0.01% of total reads, which is excellent</li>
</ul></li>
<li><strong>Barcode Accuracy</strong>:
<ul>
<li>The majority of reads (1,046,121,284) have a perfectly aligned barcode (<code>nExactMatch</code>)</li>
<li>Approximately 53 million reads show a single mismatch (<code>nMismatchOneWL</code>)</li>
<li>The absence of reads with multiple matches (<code>nMismatchToMultWL = 0</code>) suggests good barcode specificity</li>
</ul></li>
<li><strong>UMI Quality</strong>:
<ul>
<li>The number of invalid UMIs (<code>nNinUMI = 16,893,550</code>) represents a relatively small proportion of total reads and might be primarily due to sequencing errors at the beginning of reads, which is a common observation in Illumina sequencing</li>
<li>The presence of homopolymers in UMIs (<code>nUMIhomopolymer = 1,697,129</code>) is a known phenomenon that can affect molecular counting accuracy, but the relatively low number suggests this is not a major concern</li>
</ul></li>
<li><strong>Overall Matching</strong>:
<ul>
<li>The significant number of unmatched reads (<code>nNoMatch = 163,166,830</code>) suggests that a substantial portion of reads do not match expected barcodes</li>
<li>This could be due to sequencing errors or potential contamination</li>
</ul></li>
</ul>
<p>These results indicate overall good library preparation quality, with excellent cell barcode specificity, although some improvements could be made regarding UMI quality.</p>
</section>
</section>
<section id="genfull" class="level2" data-number="5.3">
<h2 data-number="5.3" class="anchored" data-anchor-id="genfull"><span class="header-section-number">5.3</span> Genfull</h2>
<div id="tbl-starsolo-features" class="quarto-layout-panel anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-starsolo-features-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5.5: STARsolo feature mapping statistics
</figcaption>
<div aria-describedby="tbl-starsolo-features-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;">Metric</th>
<th style="text-align: left;">Count</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">nUnmapped</td>
<td style="text-align: left;">114,628,761</td>
</tr>
<tr class="even">
<td style="text-align: left;">nNoFeature</td>
<td style="text-align: left;">13,153,735</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nAmbigFeature</td>
<td style="text-align: left;">936,951,032</td>
</tr>
<tr class="even">
<td style="text-align: left;">nAmbigFeatureMultimap</td>
<td style="text-align: left;">935,077,136</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nTooMany</td>
<td style="text-align: left;">0</td>
</tr>
<tr class="even">
<td style="text-align: left;">nNoExactMatch</td>
<td style="text-align: left;">125,216</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nExactMatch</td>
<td style="text-align: left;">4,471,174,098</td>
</tr>
<tr class="even">
<td style="text-align: left;">nMatch</td>
<td style="text-align: left;">971,519,404</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nMatchUnique</td>
<td style="text-align: left;">34,593,349</td>
</tr>
<tr class="even">
<td style="text-align: left;">nCellBarcodes</td>
<td style="text-align: left;">699,355</td>
</tr>
<tr class="odd">
<td style="text-align: left;">nUMIs</td>
<td style="text-align: left;">34,258,961</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</figure>
</div>
<section id="detailed-interpretation-of-starsolo-statistics" class="level3">
<h3 class="anchored" data-anchor-id="detailed-interpretation-of-starsolo-statistics">Detailed Interpretation of STARsolo Statistics</h3>
<p>Based on the official STAR documentation and explanations from Alex Dobin (STAR developer) <a href="https://github.com/alexdobin/STAR/issues/1887"><span class="citation" data-cites="alexdobin/STAR#1887"><sup><span><strong>alexdobin/STAR#1887?</strong></span></sup></span></a>, here is a detailed interpretation of our STARsolo statistics:</p>
<section id="barcode-statistics-barcodes.stats" class="level4">
<h4 class="anchored" data-anchor-id="barcode-statistics-barcodes.stats">Barcode Statistics (Barcodes.stats)</h4>
<p>Statistics with the “no” prefix indicate reads that are not used for quantification:</p>
<ul>
<li><strong>Barcode Quality</strong> :
<ul>
<li><code>nNoAdapter</code> : Reads without adapter</li>
<li><code>nNoUMI</code> : Reads without valid UMI</li>
<li><code>nNoCB</code> : Reads without valid cell barcode</li>
<li><code>nNinCB</code> : Reads with ‘N’ bases in cell barcode</li>
<li><code>nNinUMI</code> : Reads with ‘N’ bases in UMI</li>
<li><code>nUMIhomopolymer</code> : Reads with homopolymeric UMI</li>
</ul></li>
</ul>
</section>
<section id="mapping-statistics-features.stats" class="level4">
<h4 class="anchored" data-anchor-id="mapping-statistics-features.stats">Mapping Statistics (Features.stats)</h4>
<p>These statistics refer to the number of reads, except for <code>nCellBarcodes</code> and <code>nUMIs</code> which represent the number of valid cell barcodes and UMIs respectively.</p>
<ul>
<li><strong>General Mapping</strong> :
<ul>
<li><code>nUnmapped</code> : Reads not mapped to the genome</li>
<li><code>nNoFeature</code> : Reads not mapped to an annotated feature</li>
<li><code>nAmbigFeature</code> : Reads mapped to multiple features</li>
<li><code>nAmbigFeatureMultimap</code> : Subset of <code>nAmbigFeature</code> where reads are mapped to multiple genomic loci</li>
</ul></li>
<li><strong>Mapping Quality</strong> :
<ul>
<li><code>nExactMatch</code> : Reads with exact mapping</li>
<li><code>nMatch</code> : Total mapped reads (unique + multiple)</li>
<li><code>nMatchUnique</code> : Reads with unique mapping</li>
</ul></li>
</ul>
</section>
<section id="sequencing-saturation" class="level4">
<h4 class="anchored" data-anchor-id="sequencing-saturation">Sequencing Saturation</h4>
<p>Sequencing saturation is calculated as follows:</p>
<pre><code>Saturation = 1 - (N_umi / N_reads)</code></pre>
<p>where: - N_umi = number of unique CB/UMI/gene combinations - N_reads = number of reads with valid CB/UMI/gene</p>
<p>In our case, the very low saturation (0.97%) indicates that we could sequence deeper to capture more unique molecules.</p>
</section>
<section id="key-points-of-our-analysis" class="level4">
<h4 class="anchored" data-anchor-id="key-points-of-our-analysis">Key Points of Our Analysis</h4>
<ul>
<li>The high number of ambiguous mappings (<code>nAmbigFeature = 936,951,032</code>) is typical for bacterial data due to the compact nature of the genome</li>
<li>The majority of reads have exact mapping (<code>nExactMatch = 4,471,174,098</code>), indicating good mapping quality. This possibly includes both unique and multi-mapped reads that match exactly to their reference locations</li>
<li>The number of detected cell barcodes (<code>nCellBarcodes = 699,355</code>) is high, suggesting good cellular diversity</li>
<li>The number of UMIs (<code>nUMIs = 34,258,961</code>) indicates good molecular coverage</li>
</ul>
<p>These metrics suggest that our data is of good technical quality, although the low saturation indicates potential for deeper sequencing.</p>
</section>
</section>
</section>
<section id="genefull-summary-stats" class="level2" data-number="5.4">
<h2 data-number="5.4" class="anchored" data-anchor-id="genefull-summary-stats"><span class="header-section-number">5.4</span> Genefull summary stats</h2>
<div id="tbl-starsolo-summary" class="quarto-layout-panel anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-starsolo-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;5.6: STARsolo summary statistics
</figcaption>
<div aria-describedby="tbl-starsolo-summary-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: left;">Metric</th>
<th style="text-align: left;">Value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">Number of Reads</td>
<td style="text-align: left;">1,284,475,633</td>
</tr>
<tr class="even">
<td style="text-align: left;">Reads With Valid Barcodes</td>
<td style="text-align: left;">85.58%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Sequencing Saturation</td>
<td style="text-align: left;">0.97%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Q30 Bases in CB+UMI</td>
<td style="text-align: left;">95.51%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Q30 Bases in RNA read</td>
<td style="text-align: left;">95.79%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Reads Mapped to Genome: Unique+Multiple</td>
<td style="text-align: left;">89.57%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Reads Mapped to Genome: Unique</td>
<td style="text-align: left;">3.64%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Reads Mapped to GeneFull: Unique+Multiple</td>
<td style="text-align: left;">75.64%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Reads Mapped to GeneFull: Unique</td>
<td style="text-align: left;">2.69%</td>
</tr>
<tr class="even">
<td style="text-align: left;">Estimated Number of Cells</td>
<td style="text-align: left;">27,203</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Unique Reads in Cells Mapped to GeneFull</td>
<td style="text-align: left;">12,794,311</td>
</tr>
<tr class="even">
<td style="text-align: left;">Fraction of Unique Reads in Cells</td>
<td style="text-align: left;">36.98%</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Mean Reads per Cell</td>
<td style="text-align: left;">470</td>
</tr>
<tr class="even">
<td style="text-align: left;">Median Reads per Cell</td>
<td style="text-align: left;">381</td>
</tr>
<tr class="odd">
<td style="text-align: left;">UMIs in Cells</td>
<td style="text-align: left;">12,663,144</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mean UMI per Cell</td>
<td style="text-align: left;">465</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Median UMI per Cell</td>
<td style="text-align: left;">378</td>
</tr>
<tr class="even">
<td style="text-align: left;">Mean GeneFull per Cell</td>
<td style="text-align: left;">296</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Median GeneFull per Cell</td>
<td style="text-align: left;">258</td>
</tr>
<tr class="even">
<td style="text-align: left;">Total GeneFull Detected</td>
<td style="text-align: left;">6,035</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</figure>
</div>
</section>
<section id="interpretation-of-starsolo-results" class="level2" data-number="5.5">
<h2 data-number="5.5" class="anchored" data-anchor-id="interpretation-of-starsolo-results"><span class="header-section-number">5.5</span> Interpretation of STARsolo Results</h2>
<p>The STARsolo analysis revealed several key insights about our single-cell RNA-seq data:</p>
<section id="sequencing-quality-and-mapping" class="level3">
<h3 class="anchored" data-anchor-id="sequencing-quality-and-mapping">Sequencing Quality and Mapping</h3>
<ul>
<li>The sequencing quality is excellent, with over 95% of bases having Q30 quality scores in both barcode/UMI and RNA reads</li>
<li>A high proportion (85.58%) of reads contained valid cell barcodes, indicating good library preparation</li>
<li>The mapping rates are robust:
<ul>
<li>89.57% of reads mapped to the genome (unique + multiple)</li>
<li>75.64% of reads mapped to genes (unique + multiple)</li>
</ul></li>
<li>The low unique mapping rate (3.64% to genome, 2.69% to genes) is typical for bacterial RNA-seq due to …</li>
</ul>
</section>
<section id="mapping-terminology" class="level3">
<h3 class="anchored" data-anchor-id="mapping-terminology">Mapping Terminology</h3>
<ul>
<li><strong>Gene mapping</strong>: Refers to reads mapped to annotated coding sequences (CDS) only</li>
<li><strong>GeneFull mapping</strong>: Includes reads mapped to all annotated features including:
<ul>
<li>Coding sequences (CDS)</li>
<li>Untranslated regions (UTRs)</li>
<li>Non-coding RNAs</li>
<li>Intergenic regions</li>
<li>This broader mapping approach is particularly relevant for bacterial transcriptomics as it captures the full complexity of the transcriptome</li>
</ul></li>
</ul>
</section>
<section id="cell-recovery-and-expression" class="level3">
<h3 class="anchored" data-anchor-id="cell-recovery-and-expression">Cell Recovery and Expression</h3>
<ul>
<li>We estimated 27,203 cells in our dataset</li>
<li>The sequencing saturation is very low (0.97%), suggesting we could sequence deeper if needed</li>
<li>Cell-level metrics show good coverage:
<ul>
<li>Mean/median reads per cell: 470/381</li>
<li>Mean/median UMIs per cell: 465/378</li>
<li>Mean/median genes per cell: 296/258</li>
</ul></li>
<li>We detected 6,035 genes in total across all cells</li>
</ul>
</section>
<section id="data-quality-assessment" class="level3">
<h3 class="anchored" data-anchor-id="data-quality-assessment">Data Quality Assessment</h3>
<ul>
<li>The high Q30 scores and mapping rates indicate good technical quality</li>
<li>The cell-level metrics suggest sufficient coverage for downstream analysis</li>
<li>The low sequencing saturation suggests potential for deeper sequencing if needed</li>
<li>The high proportion of reads with valid barcodes (85.58%) indicates good library preparation</li>
</ul>
</section>
<section id="filter" class="level3">
<h3 class="anchored" data-anchor-id="filter">filter</h3>
<p>Nous on prend tout les barcodes pas ceux qui sont filtré donc les 820, 800 barcodes</p>
<!-- 

filtrer seulement les CDS 

r$> # show the number of cells
    ncol(seurat_object)
[1] 820800

r$> # show the number of genes
    nrow(seurat_object)
[1] 4856




 Filter the cells with less than 200 counts
    seurat_object_filtered <- subset(seurat_object, subset = nCount_RNA > 200)

r$> # Show the number of cells after filtering
    ncol(seurat_object_filtered)
[1] 65059



r$> # show the max and min of the counts
    max(seurat_object_filtered$nCount_RNA)
[1] 9452.001

r$> min(seurat_object_filtered$nCount_RNA)
[1] 200

r$> mean(seurat_object_filtered$nCount_RNA)
[1] 416.7303

r$> median(seurat_object_filtered$nCount_RNA)
[1] 323

 -->
<!-- 
r$> # show the max and min of the genes
    max(seurat_object_filtered$nFeature_RNA)
[1] 1971

r$> min(seurat_object_filtered$nFeature_RNA)
[1] 92
 seurat_object_filtered

An object of class Seurat 
4856 features across 65059 samples within 1 assay 
Active assay: RNA (4856 features, 0 variable features)
 1 layer present: counts




 -->
</section>
</section>
<section id="apres-starsolo-mettre-le-nombre-de-reads-avec-valid-barcodes-dans-la-table" class="level2" data-number="5.6">
<h2 data-number="5.6" class="anchored" data-anchor-id="apres-starsolo-mettre-le-nombre-de-reads-avec-valid-barcodes-dans-la-table"><span class="header-section-number">5.6</span> apres starsolo mettre le nombre de reads avec valid barcodes dans la table</h2>
<section id="genome-1" class="level3">
<h3 class="anchored" data-anchor-id="genome-1">genome</h3>
<p>circular representation of c-bacterial genome and read alignment voir comment faire ce type de figure et l’article qui l’avait fait</p>
</section>
<section id="section-1" class="level3">
<h3 class="anchored" data-anchor-id="section-1"></h3>
<p>Attention j’ai filtré pour ne garder que les CDS mais certains pas annotés plutot exclure les tRNA et rRNA</p>
<p>In addition, we kept the highest-scored multimapping reads, assigning a fractional count based on the number of equally good alignments, because bacterial genomes are known to contain overlapping coding sequences. We then generated a matrix of gene counts for each cell (N-by-K matrix, with N cells and K genes).</p>
<p>dans l’article de kuchina ils ont filtré les cellules en fonction du nombre de reads et de genes :</p>
<p>“Processing of data from the heat shock experiment Clustering and data analysis for the speciesmixing experiment with heat shock treatment was performed using Scanpy (59). We only kept transcriptomes that had a <strong>number of total reads higher than 200</strong>. Then, we removed the <strong>ribosomal and tRNA reads from the data</strong>, retaining only reads that represented the mRNA counts for both species. We further filtered cells based on the mRNA counts, <strong>retaining cells expressing &gt;100 reads and &gt;100 genes</strong>, and additionally filtered the genes, retaining the <strong>genes expressed in &gt;5 cells</strong>. We then applied standard Scanpy normalization and scaling, dimensionality reduction, and clustering, as described in the Scanpy tutorial (59, 60). The clusters were produced by the Louvain graph-clustering method and manually inspected for the top differentially expressed genes. After inspection, three pairs of transcriptionally similar clusters with fewer differentially expressed genes were merged, resulting in clusters 1, 2, and 3 in Fig. 1D.” (<a href="zotero://select/library/items/AWDKBVJW">Kuchina et al., 2021, p.&nbsp;8</a>) (<a href="zotero://open-pdf/library/items/YFM8QJY6?page=9&amp;annotation=LXF2XHIY">pdf</a>)</p>
</section>
</section>
</section>
<section id="initialize-the-seurat-object-with-the-raw-non-normalized-data." class="level1" data-number="6">
<h1 data-number="6"><span class="header-section-number">6</span> Initialize the Seurat object with the raw (non-normalized data).</h1>
<p>pbmc &lt;- CreateSeuratObject(counts = pbmc.data, project = “pbmc3k”, min.cells = 3, min.features = 200)</p>
<p>https://rnabioco.github.io/cellar/previous/2019/docs/2_filtering_QC.html</p>
<p>le choix de filtration sera discuté apres</p>
<p>because the reads depth is different between : Stress condition (M9F) and non-stress condition (M9) et also between the 3 timepoints (T1, T2, T3) (voir figure, reuslttas comme deja observé chez )</p>
<p>Single-cell analysis Two separate libraries were concatenated after filtering, log-normalized by 600 counts per cell, and scaled to unit variance and zero mean. Subsequently, ComBat(70) was run for batch effect correction. Normalized expression data was dimensionally reduced using principal component analysis (PCA). Shared neighbor graphs and uniform manifold approximation representations (UMAP(71)) were calculated with the first 12 principal components. All subsequent calculations were run in Python using Scanpy(72) documentation for single-cell analysis. Differential gene expression analysis Scanpy gene ranking functions (sc.tl.rank_genes_groups and sc.get.rank_genes_groups_df) were used to analyze and retrieve statistical data between two groups of interest within the annotated data object. The output parameters included names of all genes, z-score, log fold change, p-values, and adjusted p-values. To create a volcano plot from this data, either the |score| or -log(adjusted p-value) was plotted on the y-axis against the log-fold change on the x axis. Lists of defense genes from DefenseFinder(39), host response genes upregulated after phage treatment, and CPS genes were used to assign colors to each point. Host response genes were taken as the top 15 distinct genes identified in the phage-treated sample only clustering analysis relative to the untreated sample. Initial CPS gene annotations were mapped by and received from Dr.&nbsp;Laurie Comstock (University of Chicago). Calculation of co-expression scores To assess the co-expression between two genes A and B, first, probabilities of expression of individual genes were calculated as fractions of cells having above-zero normalized non-scaled expression values p(A) and p(B), respectively. Then, the probability of simultaneous expression of the two genes, p(A&amp;B), was calculated as a fraction of cells having above-zero normalized non-scaled expression values for both inspected genes. Co-expression score was calculated as a ratio of p(A&amp;B) to the multiplication of p(A) and p(B). Values close to 1 indicate independent expression of genes; values above 1 indicate co-expression, and values below 1 indicate mutually exclusive expression of genes. To define a border of a CPS operon, mean co-expression values were calculated between genes adjacent to the CPS operon and core genes of a CPS operon. Co-expression between gene sets (e.g., CPS operons) was calculated similarly to genes, with probability of expression calculated as a fraction of cells having above-zero normalized non-scaled expression values for any of genes within a set. Diffusion pseudotime analysis The data was first subsetted into the phage-treated sample alone. A root cell was defined using adata.uns[‘iroot’] = np.flatnonzero(adata.obs[‘louvan’]==0)[1], which selected a random indexed cell from the untreated cluster within all phage-treated cells. Scanpy diffusion maps were created prior to running the existing diffusion pseudotime tool with 0 branchings and 10 diffusion components. For downstream analysis using pseudotime values, 5 bins of equal size were created to group cells into pseudotime ranges (0.0-0.2, 0.2-0.4, 0.4-0.6, 0.6-0.8, and 0.8-1.0). Differential expression analysis was run to identify the top 10 distinct B. fragilis genes within each pseudotime bin, and the Clusters of Orthologous Groups (COGs) database was used to define functional categories for these genes. Duplicate genes were removed. Raw mean expression for each of these genes, grouped by functional category, was calculated using the adata.raw.X matrix.</p>
<p>[<span class="citation" data-cites="gupta"><sup><a href="06-References.html#ref-gupta" role="doc-biblioref">2</a></sup></span>]<span class="citation" data-cites="cyriaque2024"><sup><a href="06-References.html#ref-cyriaque2024" role="doc-biblioref">3</a></sup></span><span class="citation" data-cites="brettner2024a"><sup><a href="06-References.html#ref-brettner2024a" role="doc-biblioref">4</a></sup></span><span class="citation" data-cites="brettner2024b"><sup><a href="06-References.html#ref-brettner2024b" role="doc-biblioref">5</a></sup></span></p>
<p><span class="citation" data-cites="korshoj2024"><sup><a href="06-References.html#ref-korshoj2024" role="doc-biblioref">6</a></sup></span></p>
<section id="overview" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="overview"><span class="header-section-number">6.1</span> Overview</h2>
<p>This chapter presents the findings of our single-cell RNA-seq analysis of Pseudomonas, focusing on the division of labor within bacterial populations.</p>
</section>
<section id="single-cell-rna-seq-analysis" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="single-cell-rna-seq-analysis"><span class="header-section-number">6.2</span> Single-cell RNA-seq Analysis</h2>
<section id="data-quality-and-preprocessing" class="level3">
<h3 class="anchored" data-anchor-id="data-quality-and-preprocessing">Data Quality and Preprocessing</h3>
</section>
<section id="cell-type-identification" class="level3">
<h3 class="anchored" data-anchor-id="cell-type-identification">Cell Type Identification</h3>
</section>
<section id="differential-expression-analysis" class="level3">
<h3 class="anchored" data-anchor-id="differential-expression-analysis">Differential Expression Analysis</h3>
</section>
<section id="division-of-labor-patterns" class="level3">
<h3 class="anchored" data-anchor-id="division-of-labor-patterns">Division of Labor Patterns</h3>
</section>
</section>
<section id="functional-analysis" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="functional-analysis"><span class="header-section-number">6.3</span> Functional Analysis</h2>
<section id="pathway-enrichment" class="level3">
<h3 class="anchored" data-anchor-id="pathway-enrichment">Pathway Enrichment</h3>
</section>
<section id="gene-set-analysis" class="level3">
<h3 class="anchored" data-anchor-id="gene-set-analysis">Gene Set Analysis</h3>
</section>
<section id="regulatory-network-analysis" class="level3">
<h3 class="anchored" data-anchor-id="regulatory-network-analysis">Regulatory Network Analysis</h3>
</section>
</section>
<section id="integration-with-previous-studies" class="level2" data-number="6.4">
<h2 data-number="6.4" class="anchored" data-anchor-id="integration-with-previous-studies"><span class="header-section-number">6.4</span> Integration with Previous Studies</h2>
</section>
<section id="summary-of-key-findings" class="level2" data-number="6.5">
<h2 data-number="6.5" class="anchored" data-anchor-id="summary-of-key-findings"><span class="header-section-number">6.5</span> Summary of Key Findings</h2>


<div id="refs" class="references csl-bib-body" data-entry-spacing="0" data-line-spacing="2" role="list" style="display: none">
<div id="ref-kuchina2021" class="csl-entry" role="listitem">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Kuchina, A. <em>et al.</em> <a href="https://doi.org/10.1126/science.aba5257">Microbial single-cell RNA sequencing by split-pool barcoding</a>. <em>Science</em> <strong>371</strong>, eaba5257 (2021).</div>
</div>
<div id="ref-gupta" class="csl-entry" role="listitem">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Gupta, A. <em>et al.</em> Combinatorial phenotypic landscape enables bacterial resistance to phage infection. doi:<a href="https://doi.org/10.1101/2025.01.13.632860">10.1101/2025.01.13.632860</a>.</div>
</div>
<div id="ref-cyriaque2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Cyriaque, V. <em>et al.</em> <a href="https://doi.org/10.1038/s41467-024-49793-x">Single-cell RNA sequencing reveals plasmid constrains bacterial population heterogeneity and identifies a non-conjugating subpopulation</a>. <em>Nature Communications</em> <strong>15</strong>, 5853 (2024).</div>
</div>
<div id="ref-brettner2024a" class="csl-entry" role="listitem">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Brettner, L., Eder, R., Schmidlin, K. &amp; Geiler-Samerotte, K. <a href="https://doi.org/10.1002/yea.3927">An ultra high-throughput, massively multiplexable, single-cell RNA-seq platform in yeasts</a>. <em>Yeast</em> <strong>41</strong>, 242–255 (2024).</div>
</div>
<div id="ref-brettner2024b" class="csl-entry" role="listitem">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Brettner, L. &amp; Geiler-Samerotte, K. Single-cell heterogeneity in ribosome content and the consequences for the growth laws. <em>bioRxiv: The Preprint Server for Biology</em> 2024.04.19.590370 (2024) doi:<a href="https://doi.org/10.1101/2024.04.19.590370">10.1101/2024.04.19.590370</a>.</div>
</div>
<div id="ref-korshoj2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Korshoj, L. E. &amp; Kielian, T. <a href="https://doi.org/10.1038/s41467-024-54581-8">Bacterial single-cell RNA sequencing captures biofilm transcriptional heterogeneity and differential responses to immune pressure</a>. <em>Nature Communications</em> <strong>15</strong>, 10184 (2024).</div>
</div>
</div>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/02-Materials-and-methods.html" class="pagination-link" aria-label="Materials and Methods">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Materials and Methods</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/04-Discussion.html" class="pagination-link" aria-label="Discussion">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Discussion</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/vgoupille/Rapport_stage/edit/main/chapters/03-Results.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div></div></div></footer></body></html>