<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.39">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>3&nbsp; Results – Msc Bioinformatics thesis</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../site_libs/clipboard/clipboard.min.js"></script>
<script src="../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../site_libs/quarto-search/fuse.min.js"></script>
<script src="../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../">
<link href="../chapters/04-Discussion.html" rel="next">
<link href="../chapters/02-Materials-and-methods.html" rel="prev">
<script src="../site_libs/quarto-html/quarto.js"></script>
<script src="../site_libs/quarto-html/popper.min.js"></script>
<script src="../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../site_libs/quarto-html/anchor.min.js"></script>
<link href="../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../site_libs/quarto-html/quarto-syntax-highlighting-fed3a45c7a334f7a8f106de68cf0fca8.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../site_libs/bootstrap/bootstrap-aa42bd5397b87ae3001db572b235c885.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script src="../site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="../site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="../site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>
<link href="https://fonts.googleapis.com/css?family=Fira+Sans|Merriweather|Source Code Pro" rel="stylesheet">

  <script>window.backupDefine = window.define; window.define = undefined;</script><script src="https://cdn.jsdelivr.net/npm/katex@latest/dist/katex.min.js"></script>
  <script>document.addEventListener("DOMContentLoaded", function () {
 var mathElements = document.getElementsByClassName("math");
 var macros = [];
 for (var i = 0; i < mathElements.length; i++) {
  var texText = mathElements[i].firstChild;
  if (mathElements[i].tagName == "SPAN") {
   katex.render(texText.data, mathElements[i], {
    displayMode: mathElements[i].classList.contains('display'),
    throwOnError: false,
    macros: macros,
    fleqn: false
   });
}}});
  </script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@latest/dist/katex.min.css">

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="../styles/custom.css">
</head>

<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="../chapters/03-Results.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Results</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="../">Msc Bioinformatics thesis</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/vgoupille/Rapport_stage" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <a href="../thesis.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Front matter</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/01-Introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/02-Materials-and-methods.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Materials and Methods</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/03-Results.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Results</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/04-Discussion.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Discussion</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/06-References.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Bibliography</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true">
 <span class="menu-text">Appendices</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="../chapters/Appendix.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">A</span>&nbsp; <span class="chapter-title">Appendix </span></span></a>
  </div>
</li>
      </ul>
  </li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title"><b>Sections</b></h2>
   
  <ul class="collapse">
  <li><a href="#preprocessing-and-quality-control" id="toc-preprocessing-and-quality-control" class="nav-link active" data-scroll-target="#preprocessing-and-quality-control"><span class="header-section-number">3.1</span> Preprocessing and Quality Control</a></li>
  <li><a href="#sec-global-filtering" id="toc-sec-global-filtering" class="nav-link" data-scroll-target="#sec-global-filtering"><span class="header-section-number">3.2</span> Cell Quality Filtering and Dataset Characterization</a></li>
  <li><a href="#single-cell-analysis-and-transcriptional-heterogeneity" id="toc-single-cell-analysis-and-transcriptional-heterogeneity" class="nav-link" data-scroll-target="#single-cell-analysis-and-transcriptional-heterogeneity"><span class="header-section-number">3.3</span> Single-Cell Analysis and Transcriptional Heterogeneity</a></li>
  </ul>
<div class="toc-actions"><ul class="collapse"><li><a href="https://github.com/vgoupille/Rapport_stage/edit/main/chapters/03-Results.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div><div class="quarto-other-links"><h2>Other Links</h2><ul><li><a href="Oral_Valentin_GOUPILLE.pdf"><i class="bi bi-link-45deg"></i>Oral Presentation</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Results</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>This chapter presents the findings of our single-cell RNA-seq analysis of <em>P. brassicacearum</em> R401. We first established data quality through preprocessing and quality control steps, then applied filtering strategies to ensure reliable cell identification and gene expression quantification.</p>
<section id="preprocessing-and-quality-control" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="preprocessing-and-quality-control"><span class="header-section-number">3.1</span> Preprocessing and Quality Control</h2>
<section id="trimming-and-quality-control-of-fastq-files" class="level3">
<h3 class="anchored" data-anchor-id="trimming-and-quality-control-of-fastq-files">Trimming and Quality Control of FASTQ Files</h3>
<div id="fig-adaptateur" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-adaptateur-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-adaptateur" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-before" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-before-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/adaptateur_R1_BC76_bf_trimm.png" class="lightbox" data-gallery="fig-adaptateur" title="Figure&nbsp;3.1&nbsp;(a): before trimming"><img src="../figures/adaptateur_R1_BC76_bf_trimm.png" class="img-fluid figure-img" data-ref-parent="fig-adaptateur"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-before-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) before trimming
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-adaptateur" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-after" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-after-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/adaptateur_R1_BC76_after_trimm.png" class="lightbox" data-gallery="fig-adaptateur" title="Figure&nbsp;3.1&nbsp;(b): after trimming"><img src="../figures/adaptateur_R1_BC76_after_trimm.png" class="img-fluid figure-img" data-ref-parent="fig-adaptateur"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-after-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) after trimming
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-adaptateur-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.1: Artifacts content of the sublibrary BC_0076 before and after trimming with Cutadapt and Fastp
</figcaption>
</figure>
</div>
<p><em>The adapter content analysis shows the presence of various sequencing artifacts in the raw data (example of BC_0076). Before trimming (<a href="#fig-before" class="quarto-xref">Figure&nbsp;<span>3.1 (a)</span></a>), distinct patterns are visible: polyG stretches (pink), polyA tails (cyan), and Illumina adapter sequences (purple). After trimming (<a href="#fig-after" class="quarto-xref">Figure&nbsp;<span>3.1 (b)</span></a>), these artifacts are effectively removed, resulting in clean sequence data suitable for downstream analysis.</em></p>
<div id="tbl-summary-trimming" class="quarto-layout-panel anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-summary-trimming-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3.1: Summary of sequence metrics before and after trimming of each sublibrary with Cutadapt and Fastp (R1 read length metrics)
</figcaption>
<div aria-describedby="tbl-summary-trimming-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<table class="caption-top table">
<colgroup>
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
<col style="width: 16%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Sample Name</th>
<th style="text-align: left;">BC_0076</th>
<th style="text-align: left;">BC_0077</th>
<th style="text-align: left;">BC_0079</th>
<th style="text-align: left;">BC_0080</th>
<th style="text-align: left;"><span class="math inline">Mean</span>/Total</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">R1 length before trimming</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;">241bp</td>
<td style="text-align: left;"><span class="math inline">241bp</span></td>
</tr>
<tr class="even">
<td style="text-align: left;">R1 length after trimming</td>
<td style="text-align: left;">127bp</td>
<td style="text-align: left;">157bp</td>
<td style="text-align: left;">152bp</td>
<td style="text-align: left;">132bp</td>
<td style="text-align: left;"><span class="math inline">142bp</span></td>
</tr>
<tr class="odd">
<td style="text-align: left;">R1 number of sequences before trimming</td>
<td style="text-align: left;">631.4M</td>
<td style="text-align: left;">325.5M</td>
<td style="text-align: left;">379.1M</td>
<td style="text-align: left;">397.7M</td>
<td style="text-align: left;"><strong>1733.7M</strong></td>
</tr>
<tr class="even">
<td style="text-align: left;">R1 number of sequences after trimming</td>
<td style="text-align: left;">450.8M</td>
<td style="text-align: left;">248.4M</td>
<td style="text-align: left;">285.2M</td>
<td style="text-align: left;">300.1M</td>
<td style="text-align: left;"><strong>1284.5M</strong></td>
</tr>
<tr class="odd">
<td style="text-align: left;">Change in number of sequences</td>
<td style="text-align: left;">-28.6%</td>
<td style="text-align: left;">-23.7%</td>
<td style="text-align: left;">-24.8%</td>
<td style="text-align: left;">-24.5%</td>
<td style="text-align: left;"><strong>-25.4%</strong></td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</figure>
</div>
<p>The trimming process successfully removed various sequencing artifacts and improved data quality for downstream analysis. The preprocessing pipeline eliminated template-switching oligo (TSO) sequences, polyG stretches (NovaSeq-specific artifacts), polyA tails, and adapter sequences from the raw sequencing data (<a href="#fig-adaptateur" class="quarto-xref">Figure&nbsp;<span>3.1</span></a>, <a href="#tbl-summary-trimming" class="quarto-xref">Table&nbsp;<span>3.1</span></a>). This cleaning step was essential for accurate gene expression quantification and reliable single-cell analysis. The trimming process resulted in an average reduction of 25.4% in the total number of sequences across all sublibraries, from 1,733.7 million to 1,284.5 million reads (<a href="#tbl-summary-trimming" class="quarto-xref">Table&nbsp;<span>3.1</span></a>).</p>
</section>
<section id="alignment-and-quantification-with-starsolo" class="level3">
<h3 class="anchored" data-anchor-id="alignment-and-quantification-with-starsolo">Alignment and Quantification with STARsolo</h3>
<p>Following the successful preprocessing and trimming of our sequencing data, we performed alignment against the reference genome of <em>P. brassicacearum R401</em> (<a href="#fig-genome" class="quarto-xref">Figure&nbsp;<span>3.2 (a)</span></a>). The barcode reading and quantification by STARsolo<span class="citation" data-cites="dobin2013 kaminow"><sup><a href="06-References.html#ref-dobin2013" role="doc-biblioref">1</a>,<a href="06-References.html#ref-kaminow" role="doc-biblioref">2</a></sup></span> provided comprehensive quality metrics and mapping statistics.</p>
<div id="tbl-starsolo-comprehensive" class="quarto-layout-panel anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-starsolo-comprehensive-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;3.2: Summary of metrics obtained with STARsolo after barcode-UMI reading and alignment with the reference genome of P. brassicacearum R401
</figcaption>
<div aria-describedby="tbl-starsolo-comprehensive-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell" style="flex-basis: 100.0%;justify-content: center;">
<table class="caption-top table">
<colgroup>
<col style="width: 66%">
<col style="width: 33%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: left;">Metric</th>
<th style="text-align: left;">Count/Percentage</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><strong>Total Reads Processed</strong></td>
<td style="text-align: left;">1,284,475,633</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>N_reads with valid barcodes</strong></td>
<td style="text-align: left;">1,099,327,755 (<strong>85.58%</strong>)</td>
</tr>
<tr class="odd">
<td style="text-align: left;">- Exact Barcode Match</td>
<td style="text-align: left;">1,046,121,284 (<strong>81.44%</strong>)</td>
</tr>
<tr class="even">
<td style="text-align: left;">- Single Mismatch Barcode</td>
<td style="text-align: left;">53,206,471 (<strong>4.14%</strong>)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong>Q30 Bases in RNA reads</strong></td>
<td style="text-align: left;">95.79%</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>Q30 Bases in CB+UMI</strong></td>
<td style="text-align: left;">95.51%</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong>Unique Gene Mapping</strong></td>
<td style="text-align: left;">34,593,349 (2.69%)</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>Unique + Multiple Gene Mapping</strong></td>
<td style="text-align: left;">971,519,404 (75.64%)</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong>Total Genes Detected</strong></td>
<td style="text-align: left;">6,035 on the total of 6,249 genes (96.57%)</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>Cell Barcodes Detected</strong></td>
<td style="text-align: left;">699,355</td>
</tr>
<tr class="odd">
<td style="text-align: left;"><strong>N_umi</strong></td>
<td style="text-align: left;">36,565,214</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</figure>
</div>
<p>The STARsolo analysis demonstrated excellent barcode quality with 85.58% of reads having valid barcodes, falling within the expected range of 70-90% for successful experiments<span class="citation" data-cites="gaisser2024"><sup><a href="06-References.html#ref-gaisser2024" role="doc-biblioref">3</a></sup></span>. The sequencing quality was outstanding with Q30 scores above 95% for both RNA reads and barcode/UMI sequences.</p>
<p>Gene mapping analysis revealed that 75.64% of reads mapped to genes, with 2.69% showing unique mapping, which is typical for bacterial genomes with overlapping genes and repeated sequences<span class="citation" data-cites="gaisser2024"><sup><a href="06-References.html#ref-gaisser2024" role="doc-biblioref">3</a></sup></span>. The fraction of uniquely aligned reads falls within the expected range of 3-12% for bacterial samples<span class="citation" data-cites="gaisser2024"><sup><a href="06-References.html#ref-gaisser2024" role="doc-biblioref">3</a></sup></span>. A total of 6,035 genes were detected across 699,355 unique cell barcodes, representing 96.6% of the 6,249 genes annotated in the <em>P. brassicacearum R401</em> genome.</p>
<div id="fig-genome-transcriptome" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-genome-transcriptome-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-genome-transcriptome" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-genome" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-genome-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/genome_description.png" class="lightbox" data-gallery="fig-genome-transcriptome" title="Figure&nbsp;3.2&nbsp;(a): genome"><img src="../figures/genome_description.png" class="img-fluid figure-img" data-ref-parent="fig-genome-transcriptome"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-genome-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) genome
</figcaption>
</figure>
</div>
</div>
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-genome-transcriptome" style="flex-basis: 50.0%;justify-content: flex-start;">
<div id="fig-transcriptome" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-transcriptome-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/transcriptome_description.png" class="lightbox" data-gallery="fig-genome-transcriptome" title="Figure&nbsp;3.2&nbsp;(b): transcriptome"><img src="../figures/transcriptome_description.png" class="img-fluid figure-img" data-ref-parent="fig-genome-transcriptome"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-transcriptome-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) transcriptome
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-genome-transcriptome-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.2: Compositional analysis of <em>P. brassicacearum</em> R401 genome annotation and transcriptome distribution of different RNA types
</figcaption>
</figure>
</div>
<section id="genome-and-transcriptome-composition" class="level4">
<h4 class="anchored" data-anchor-id="genome-and-transcriptome-composition">Genome and Transcriptome Composition</h4>
<p>The genome and transcriptome composition analysis reveals the distribution of different RNA types in our <em>P. brassicacearum</em> sample (<a href="#fig-genome-transcriptome" class="quarto-xref">Figure&nbsp;<span>3.2</span></a>). In the <a href="https://www.ncbi.nlm.nih.gov/datasets/gene/GCA_030064105.1/">annotated</a> genome of <em>P. brassicacearum</em> R401, we identified 6,100 mRNA genes (97.6% of total genes), 65 tRNA genes (1.1% of total genes), 16 rRNA genes (0.3%) and 68 other genes. However, in the transcriptome analysis of our 36,565,214 UMIs <a href="#tbl-starsolo-comprehensive" class="quarto-xref">Table&nbsp;<span>3.2</span></a>, the distribution shows a different pattern: 57.1% correspond to mRNA transcripts (21 millions), while 28.4% are rRNA transcripts (representing approximately 10 millions UMIs), 10.2% are tRNA transcripts (representing approximately 4 millions UMIs) and 4.4% are other genes (representing approximately 1.5 millions UMIs). This distribution concords with the well-established observation that rRNA and tRNA transcripts represent a large proportion of the bacterial transcriptome despite constituting only a small fraction of the genome, due to their high transcriptional activity and stability<span class="citation" data-cites="nishimura2025"><sup><a href="06-References.html#ref-nishimura2025" role="doc-biblioref">4</a></sup></span>.</p>
</section>
</section>
</section>
<section id="sec-global-filtering" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="sec-global-filtering"><span class="header-section-number">3.2</span> Cell Quality Filtering and Dataset Characterization</h2>
<p>STARsolo<span class="citation" data-cites="dobin2013 kaminow"><sup><a href="06-References.html#ref-dobin2013" role="doc-biblioref">1</a>,<a href="06-References.html#ref-kaminow" role="doc-biblioref">2</a></sup></span> detected 699,355 out of 820,800 possible barcode combinations, significantly more than the approximately 3,000 cells targeted in our sublibrary design <a href="#tbl-starsolo-comprehensive" class="quarto-xref">Table&nbsp;<span>3.2</span></a>. This large discrepancy indicates the presence of contaminating barcodes that can occur throughout the protocol, including potential contamination in the source oligonucleotide plates. To proceed with single-cell analysis, we needed to identify and retain only genuine cells among this large barcode population. STARsolo employs a knee plot strategy to identify “real” cells<span class="citation" data-cites="luecken2019"><sup><a href="06-References.html#ref-luecken2019" role="doc-biblioref">5</a></sup></span>. However, as recommended in the literature (Gaisser et al.&nbsp;2024<span class="citation" data-cites="gaisser2024"><sup><a href="06-References.html#ref-gaisser2024" role="doc-biblioref">3</a></sup></span>), we chose to apply our own filtering criteria rather than using STARsolo’s default KneePant method, which would have resulted in approximately 27,000 cells. This decision was based on the consideration that our experimental design included cells grown under different conditions (culture medium composition) and at different growth stages, which could result in varying transcriptional activity levels. Therefore, applying a single threshold across all conditions might eliminate cells with naturally lower but biologically relevant expression profiles. Instead, we chose to apply different thresholds to retain the best cells from each biological replicate, ensuring balanced representation across experimental conditions. This approach was validated by our initial UMI-based filtering results <a href="Appendix.html#sec-appendix-cell-filtering" class="quarto-xref"><span>Section A.4</span></a>, which revealed clear differences between culture conditions with M9 medium (nutrient-limited) showing fewer retained barcodes compared to M9F (nutrient-rich) when we used a global threshold of 100 UMIs per cell, confirming that different thresholds per samples were indeed necessary to capture cells from all experimental conditions.</p>
<div id="fig-final_filtering" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-final_filtering-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-final_filtering" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-splitplot-barcodes-final" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-splitplot-barcodes-final-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/mRNA__Genes_barplot.png" class="lightbox" data-gallery="fig-final_filtering" title="Figure&nbsp;3.3&nbsp;(a): Number of cell barcodes retained for each biological replicate (with technical replicates indicated within each bar) across different culture conditions and timepoints after final filtering"><img src="../figures/mRNA__Genes_barplot.png" class="img-fluid figure-img" data-ref-parent="fig-final_filtering"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-splitplot-barcodes-final-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Number of cell barcodes retained for each biological replicate (with technical replicates indicated within each bar) across different culture conditions and timepoints after final filtering
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-final_filtering" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-filtered_mRNA_violin_UMI" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-filtered_mRNA_violin_UMI-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/mRNA_GenesUMIs_violin.png" class="lightbox" data-gallery="fig-final_filtering" title="Figure&nbsp;3.3&nbsp;(b): Distribution of UMI counts for mRNA genes per cell across culture conditions after final filtering (with technical replicates indicated by specific colors)"><img src="../figures/mRNA_GenesUMIs_violin.png" class="img-fluid figure-img" data-ref-parent="fig-final_filtering"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-filtered_mRNA_violin_UMI-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Distribution of UMI counts for mRNA genes per cell across culture conditions after final filtering (with technical replicates indicated by specific colors)
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-final_filtering" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-filtered_mRNA_violin_gene" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-filtered_mRNA_violin_gene-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/mRNAGenesNumber_violin.png" class="lightbox" data-gallery="fig-final_filtering" title="Figure&nbsp;3.3&nbsp;(c): Distribution of mRNA gene numbers per cell across culture conditions after final filtering (with technical replicates indicated by specific colors)"><img src="../figures/mRNAGenesNumber_violin.png" class="img-fluid figure-img" data-ref-parent="fig-final_filtering"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-filtered_mRNA_violin_gene-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(c) Distribution of mRNA gene numbers per cell across culture conditions after final filtering (with technical replicates indicated by specific colors)
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-final_filtering" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-scatter_final" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-scatter_final-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/scatter_final.png" class="lightbox" data-gallery="fig-final_filtering" title="Figure&nbsp;3.3&nbsp;(d): Relationship between UMI counts and mRNA gene counts per cell in the final filtered dataset"><img src="../figures/scatter_final.png" class="img-fluid figure-img" data-ref-parent="fig-final_filtering"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-scatter_final-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(d) Relationship between UMI counts and mRNA gene counts per cell in the final filtered dataset
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-final_filtering-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.3: Metrics of filtered cells across experimental conditions (Culture Medium × Biological Replicate × Sampling Time) in the single-cell RNA-seq experiment.
</figcaption>
</figure>
</div>
<p>Following the initial UMI-based filtering, additional quality control steps were applied including doublet removal, biological replicate-based filtering, and mRNA type filtering. The final dataset contained approximately 3,000 cells, representing the highest-quality cells selected from each biological replicate to ensure balanced representation across experimental conditions. The filtering by biological replicate shows that some technical replicates were partially or completely eliminated to retain only the best-performing ones. The distribution of retained cells shows approximately 160 cells per biological replicate (<a href="#fig-splitplot-barcodes-final" class="quarto-xref">Figure&nbsp;<span>3.3 (a)</span></a>). When few technical replicate categories remain for a biological replicate, it indicates high variability between technical replicates, likely due to differential barcode efficiency since each technical replicate possesses different barcode 1 combinations (<a href="#fig-splitplot-barcodes-final" class="quarto-xref">Figure&nbsp;<span>3.3 (a)</span></a>). Some technical replicates were more efficient than others (e.g., M9F A sampling T2 had one technical replicate that was much more efficient than its group and therefore almost the only one for which cells passed the filter). Following this filtering, technical replicates are no longer considered for subsequent analyses, and we focus on the biological replicate level.</p>
<p>The following visualizations also show the distribution of mRNA gene expression patterns across the filtered cell population, specifically the number of mRNA genes detected per cell and the number of UMIs associated with mRNA genes per cell <a href="#fig-filtered_mRNA_violin_UMI" class="quarto-xref">Figure&nbsp;<span>3.3 (b)</span></a>, <a href="#fig-filtered_mRNA_violin_gene" class="quarto-xref">Figure&nbsp;<span>3.3 (c)</span></a>, <a href="#fig-scatter_final" class="quarto-xref">Figure&nbsp;<span>3.3 (d)</span></a>.</p>
<p>M9F medium consistently showed higher values compared to M9 (<a href="#fig-scatter_final" class="quarto-xref">Figure&nbsp;<span>3.3 (d)</span></a>), with mean mRNA counts of 749 for M9F versus 372 for M9, and mean gene counts of 462 for M9F versus 270 for M9.</p>
<p>Biological replicate variability was particularly pronounced at T1 for both conditions and at T3 for M9 conditions <a href="#fig-filtered_mRNA_violin_UMI" class="quarto-xref">Figure&nbsp;<span>3.3 (b)</span></a>, <a href="#fig-filtered_mRNA_violin_gene" class="quarto-xref">Figure&nbsp;<span>3.3 (c)</span></a>. Under nutrient-limited conditions (M9), cells showed markedly reduced transcriptional activity at T2 and T3 timepoints, with some replicates displaying very low UMI and gene counts, suggesting either reduced cellular activity or technical challenges in recovering stressed cells. The scatter plot (<a href="#fig-scatter_final" class="quarto-xref">Figure&nbsp;<span>3.3 (d)</span></a>) reveals a strong positive correlation between UMI counts and mRNA gene numbers, confirming the quality of the filtered dataset and the relationship between sequencing depth and gene detection.</p>
<p>These results demonstrate that the filtering strategy effectively removed poor-quality barcodes to retain only potential “real cells” from each biological replicate that can be analyzed for downstream investigation.</p>
</section>
<section id="single-cell-analysis-and-transcriptional-heterogeneity" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="single-cell-analysis-and-transcriptional-heterogeneity"><span class="header-section-number">3.3</span> Single-Cell Analysis and Transcriptional Heterogeneity</h2>
<p>Following the quality filtering steps, we performed dimensionality reduction and clustering analysis to explore transcriptional heterogeneity across cell populations and assess global differences between culture conditions. The analysis began with data transformation using the BacSC method (see <a href="Appendix.html#fig-pca-choose" class="quarto-xref">Figure&nbsp;<span>A.3 (b)</span></a> for detailed about it), followed by Principal Component Analysis (PCA) to identify the optimal number of dimensions for downstream analysis. UMAP visualization and clustering were then used to detect global transcriptional differences between experimental conditions.</p>
<section id="principal-component-analysis" class="level3">
<h3 class="anchored" data-anchor-id="principal-component-analysis">Principal Component Analysis</h3>
<div id="fig-single-cell-PCA" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-single-cell-PCA-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-single-cell-PCA" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-pca-condition" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-pca-condition-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/fig_sc/pca_condition.png" class="lightbox" data-gallery="fig-single-cell-PCA" title="Figure&nbsp;3.4&nbsp;(a): Unsupervised PCA projection highlighting transcriptional heterogeneity across experimental conditions. Cells are colored by culture medium, biological replicate (grouped or not) at specific timepoint, and total mRNA contents. (i.e.&nbsp;nomenclature: M9F_A_T1 corresponds to biological replicate A of M9F at sampling time 1)"><img src="../figures/fig_sc/pca_condition.png" class="img-fluid figure-img" data-ref-parent="fig-single-cell-PCA"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-pca-condition-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) Unsupervised PCA projection highlighting transcriptional heterogeneity across experimental conditions. Cells are colored by culture medium, biological replicate (grouped or not) at specific timepoint, and total mRNA contents. (i.e.&nbsp;nomenclature: M9F_A_T1 corresponds to biological replicate A of M9F at sampling time 1)
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-single-cell-PCA" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-pca1-summary-top" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-pca1-summary-top-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/fig_sc/pca1_summary_top.png" class="lightbox" data-gallery="fig-single-cell-PCA" title="Figure&nbsp;3.4&nbsp;(b): Heatmap of the top contributing genes (top 20) to the first principal components (PC1), based on scaled expression values (z-score per gene). Rows represent genes, columns represent single cells collected at each timepoints for all biological replicates. The legend at the right side of the heatmap shows the full name of the genes."><img src="../figures/fig_sc/pca1_summary_top.png" class="img-fluid figure-img" data-ref-parent="fig-single-cell-PCA"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-pca1-summary-top-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) Heatmap of the top contributing genes (top 20) to the first principal components (PC1), based on scaled expression values (z-score per gene). Rows represent genes, columns represent single cells collected at each timepoints for all biological replicates. The legend at the right side of the heatmap shows the full name of the genes.
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-single-cell-PCA-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.4: Principal Component Analysis (PCA) results showing transcriptional heterogeneity and gene contribution patterns in <em>P. brassicacearum</em> populations
</figcaption>
</figure>
</div>
<p>The PCA analysis revealed a characteristic pattern of low variance explained by individual principal components (<a href="Appendix.html#fig-pca-choose" class="quarto-xref">Figure&nbsp;<span>A.3 (b)</span></a>), which is typical for bacterial single-cell RNA-seq data, particularly under stress conditions. PC1 captured approximately 0.6% of the total variance and dominated the other components, which showed even lower contributions. This indicates that the transcriptional variability is diffuse or weakly structured in the dataset. In this context, for the next analysis we chose to retain the first 5 PCs for UMAP embedding to capture the few potentially meaningful signals carried by the leading components while avoiding the integration of excessive noise. This selection provides a balance between maintaining minimal expressiveness for dimensionality reduction and exercising caution given the low inherent structure of the data.</p>
<p>PC1 successfully distinguished between the two culture conditions (M9 and M9F), revealing distinct transcriptional patterns (<a href="#fig-pca-condition" class="quarto-xref">Figure&nbsp;<span>3.4 (a)</span></a>). In fact, we can see that M9 T2 and M9 T3 samples (brown and purple points in the Sampling Time facet, top right) were clearly separated from other conditions, while M9 T1 showed no significant difference from M9F conditions. We can also observe this on the heatmaps (<a href="#fig-pca1-summary-top" class="quarto-xref">Figure&nbsp;<span>3.4 (b)</span></a>).</p>
<p>Notably, M9 samples positioned on the left side of PC1 showed greater dispersion compared to those on the right. Furthermore, PC1 axis alignment followed a gradient of total mRNA concentration per cell, suggesting that this component primarily captures differences in overall transcriptional activity between conditions.</p>
<section id="gene-contribution-analysis-to-pc1" class="level4">
<h4 class="anchored" data-anchor-id="gene-contribution-analysis-to-pc1">Gene Contribution Analysis to PC1</h4>
<p>The heatmap analysis with z-score scaling (ranging from 0 to 1) revealed the genes contributing most significantly to PC1 variance (<a href="#fig-pca1-summary-top" class="quarto-xref">Figure&nbsp;<span>3.4 (b)</span></a>). Each row represents a gene and each column represents a cell, with the legend at the bottom corresponding to biological replicates.</p>
<p>The genes contributing most to PC1 variance were primarily associated with translational regulation and iron metabolism, reflecting the stress response to iron limitation in M9 medium. Among the translational regulators, we identified rpsA (30S ribosomal protein S1), raiA (ribosome-associated translation inhibitor), and htpG (HSP90 protein chaperone). HtpG, as a molecular chaperone, plays an indirect but essential role in the biosynthesis and functionality of siderophores. For example, HtpG facilitates the correct folding and stability of enzymes involved in siderophore synthesis, such as those participating in yersiniabactine production in <em>Yersinia</em> species<span class="citation" data-cites="mangla"><sup><a href="06-References.html#ref-mangla" role="doc-biblioref">6</a></sup></span>.Notably, porphobilinogen synthase was also identified among the top 20 genes contributing to PC1 variance. This finding is particularly relevant given our experimental design using iron-limited M9 medium. In bacteria, the link between porphobilinogen (PBG) and iron is essential for heme synthesis, a cofactor indispensable for many proteins involved in metabolism, cellular respiration, and electron transport. The biosynthesis of heme in bacteria follows a pathway similar to eukaryotes, with PBG being a key intermediate formed by the condensation of two delta-aminolevulinic acid (ALA) molecules, catalyzed by porphobilinogen synthase. This process is regulated by iron availability in bacterial cells. The differential expression of porphobilinogen synthase between M9 and M9F conditions thus reflects the cellular adaptation to iron stress<span class="citation" data-cites="choby2016"><sup><a href="06-References.html#ref-choby2016" role="doc-biblioref">7</a></sup></span>. The analysis also identified TonB-dependent siderophore receptors, which are essential for high-affinity iron acquisition in gram-negative bacteria<span class="citation" data-cites="moeck1998 fujita2019"><sup><a href="06-References.html#ref-moeck1998" role="doc-biblioref">8</a>,<a href="06-References.html#ref-fujita2019" role="doc-biblioref">9</a></sup></span>. These outer membrane-localized proteins bind iron chelates at the cell surface and promote their uptake. Together with porphobilinogen synthase, these genes form a coherent transcriptional response to iron limitation, explaining the clear separation of M9 conditions along PC1.</p>
<p>While PC1 primarily captured differences in transcriptional activity and stress response, PC2 revealed distinct patterns related to cellular motility and chemotaxis (<a href="Appendix.html#fig-pca2-summary-top" class="quarto-xref">Figure&nbsp;<span>A.3 (d)</span></a>). For example, methyl-accepting chemotaxis proteins (MCPs) and flagellin genes were identified as major contributors to PC2 variance, suggesting a role in chemotaxis and motility responses to environmental conditions.<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></p>
</section>
</section>
<section id="umap-visualization-and-clustering-analysis" class="level3">
<h3 class="anchored" data-anchor-id="umap-visualization-and-clustering-analysis">UMAP Visualization and Clustering Analysis</h3>
<p>Following the PCA analysis, we performed UMAP visualization and clustering analysis to explore transcriptional heterogeneity across cell populations and assess global differences between culture conditions.</p>
<p>The clustering analysis with resolution 0.1 identified two main clusters, while resolution 0.25 revealed three clusters (<a href="#fig-gene-cluster" class="quarto-xref">Figure&nbsp;<span>3.5 (b)</span></a>). Both clustering approaches successfully separated the experimental conditions, with M9 T2 and M9 T3 samples clearly distinguished from other conditions. This separation aligns with the PCA results, confirming the distinct transcriptional profiles of cells under iron-limited conditions at later timepoints.</p>
<section id="differential-expression-analysis-between-clusters" class="level4">
<h4 class="anchored" data-anchor-id="differential-expression-analysis-between-clusters">Differential Expression Analysis Between Clusters</h4>
<p>To characterize the transcriptional differences between the identified clusters, we performed differential expression analysis between the two clusters obtained with resolution 0.1. We used the Wilcoxon rank-sum test with FDR correction (p &lt; 0.05) to identify significantly differentially expressed genes. The complete results of this analysis are presented in the appendix (<a href="Appendix.html#fig-diff-exp" class="quarto-xref">Figure&nbsp;<span>A.3 (a)</span></a>).</p>
<p>Group 1 (primarily M9 T2 and T3 cells) exhibited globally lower expression levels compared to Group 0 (primarily M9F and M9 T1 cells), which showed a more heterogeneous expression gradient (<a href="#fig-gene-cluster" class="quarto-xref">Figure&nbsp;<span>3.5 (b)</span></a>). This pattern reflects the reduced transcriptional activity observed in cells under iron stress conditions.</p>
</section>
<section id="characterization-of-differentially-expressed-genes" class="level4">
<h4 class="anchored" data-anchor-id="characterization-of-differentially-expressed-genes">Characterization of Differentially Expressed Genes</h4>
<p>Among the significantly differentially expressed genes, we identified three representative examples that illustrate the distinct transcriptional programs between clusters. The <em>RplA</em> gene that codes for ribosomal protein RplA (30S ribosomal protein L1) showed lower expression in M9 T2 and T3 cells compared to M9F and M9 T1 cells. RplA is constitutively expressed as it is essential for protein synthesis, with expression levels reflecting the cell’s translational activity. Higher expression typically indicates active protein synthesis during rapid growth, while lower expression suggests reduced translational activity under stress conditions. The <em>QLH64-28090</em> (phasin) gene showed an opposite pattern, with higher expression in M9 T2 and T3 cells. Phasins are multifunctional proteins associated with polyhydroxyalkanoate (PHA) granules that play crucial roles in stress response and energy metabolism<span class="citation" data-cites="mezzina2016"><sup><a href="06-References.html#ref-mezzina2016" role="doc-biblioref">10</a></sup></span>. They form an interface between the hydrophobic PHA granules and the hydrophilic cytoplasm, regulating both PHA accumulation and utilization. Under stress conditions, phasins can activate PHA depolymerization to release energy metabolites, increase PHA synthase activity . Additionally, some phasins exhibit chaperone-like properties, protecting cellular proteins against stress-induced denaturation and oxidative damage. The <em>Flagellin</em> gene, which is a housekeeping gene, was the most highly expressed gene in the raw data matrix, also showed differential expression between clusters, with higher expression in M9 T2 and T3 cells compared to M9F and M9 T1.</p>
</section>
<section id="global-expression-patterns-and-cluster-heterogeneity" class="level4">
<h4 class="anchored" data-anchor-id="global-expression-patterns-and-cluster-heterogeneity">Global Expression Patterns and Cluster Heterogeneity</h4>
<p>However, the analysis revealed significant heterogeneity within each cluster, with groups of cells showing distinct expression patterns for specific genes compared to the global cluster average. This heterogeneity is particularly evident in the violin plots presented in the appendix <a href="Appendix.html#fig-violin-sc" class="quarto-xref">Figure&nbsp;<span>A.3 (c)</span></a>, which show variable distributions of gene expression within clusters. In the heatmap <a href="Appendix.html#fig-heatmap" class="quarto-xref">Figure&nbsp;<span>A.3 (b)</span></a> representing top significant differentially expressed genes, we can see that some genes are expressed at very high levels, particularly among the significant genes for cluster 1. This observation could suggest the presence of specialized subpopulations within the broader transcriptional groups, potentially indicating fine-grained division of labor mechanisms at the cellular level.</p>
<div id="fig-single-cell-analysis" class="quarto-layout-panel">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-single-cell-analysis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-single-cell-analysis" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-umap" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-umap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/fig_sc/umap.png" class="lightbox" data-gallery="fig-single-cell-analysis" title="Figure&nbsp;3.5&nbsp;(a): UMAP visualization of the single-cell RNA-seq data, colored by culture medium, biological replicate (grouped or not) at specific timepoint, and total mRNA contents. Leiden clustering with a resolution of 0.1 and 0.25 are represented in two bottoms graphs."><img src="../figures/fig_sc/umap.png" class="img-fluid figure-img" data-ref-parent="fig-single-cell-analysis"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-umap-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(a) UMAP visualization of the single-cell RNA-seq data, colored by culture medium, biological replicate (grouped or not) at specific timepoint, and total mRNA contents. Leiden clustering with a resolution of 0.1 and 0.25 are represented in two bottoms graphs.
</figcaption>
</figure>
</div>
</div>
</div>
<div class="quarto-layout-row">
<div class="quarto-layout-cell-subref quarto-layout-cell" data-ref-parent="fig-single-cell-analysis" style="flex-basis: 100.0%;justify-content: flex-start;">
<div id="fig-gene-cluster" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-subfloat-fig figure">
<div aria-describedby="fig-gene-cluster-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="../figures/fig_sc/gene_cluster.png" class="lightbox" data-gallery="fig-single-cell-analysis" title="Figure&nbsp;3.5&nbsp;(b): UMAP visualization colored by expression of examples top differentially expressed genes (FDR < 0.05) between clusters: RplA (30S ribosomal protein L1), QLH64-28090 (phasin), and flagellin. Color intensity indicates relative expression levels (yellow: high expression, blue: low expression)."><img src="../figures/fig_sc/gene_cluster.png" class="img-fluid figure-img" data-ref-parent="fig-single-cell-analysis"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-subfloat-caption quarto-subfloat-fig" id="fig-gene-cluster-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
(b) UMAP visualization colored by expression of examples top differentially expressed genes (FDR &lt; 0.05) between clusters: RplA (30S ribosomal protein L1), QLH64-28090 (phasin), and flagellin. Color intensity indicates relative expression levels (yellow: high expression, blue: low expression).
</figcaption>
</figure>
</div>
</div>
</div>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-single-cell-analysis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3.5: Single-cell RNA-seq analysis results showing gene expression patterns and cellular heterogeneity in <em>P. brassicacearum</em> populations
</figcaption>
</figure>
</div>


<div id="refs" class="references csl-bib-body" data-entry-spacing="0" data-line-spacing="2" role="list" style="display: none">
<div id="ref-dobin2013" class="csl-entry" role="listitem">
<div class="csl-left-margin">1. </div><div class="csl-right-inline">Dobin, A. <em>et al.</em> <a href="https://doi.org/10.1093/bioinformatics/bts635">STAR: ultrafast universal RNA-seq aligner</a>. <em>Bioinformatics (Oxford, England)</em> <strong>29</strong>, 15–21 (2013).</div>
</div>
<div id="ref-kaminow" class="csl-entry" role="listitem">
<div class="csl-left-margin">2. </div><div class="csl-right-inline">Kaminow, B., Yunusov, D. &amp; Dobin, A. STARsolo: accurate, fast and versatile mapping/quantification of single-cell and single-nucleus RNA-seq data. doi:<a href="https://doi.org/10.1101/2021.05.05.442755">10.1101/2021.05.05.442755</a>.</div>
</div>
<div id="ref-gaisser2024" class="csl-entry" role="listitem">
<div class="csl-left-margin">3. </div><div class="csl-right-inline">Gaisser, K. D. <em>et al.</em> <a href="https://doi.org/10.1038/s41596-024-01007-w">High-throughput single-cell transcriptomics of bacteria using combinatorial barcoding</a>. <em>Nature Protocols</em> <strong>19</strong>, 3048–3084 (2024).</div>
</div>
<div id="ref-nishimura2025" class="csl-entry" role="listitem">
<div class="csl-left-margin">4. </div><div class="csl-right-inline">Nishimura, M., Takahashi, K. &amp; Hosokawa, M. Recent advances in single-cell RNA sequencing of bacteria: Techniques, challenges, and applications. <em>Journal of Bioscience and Bioengineering</em> (2025) doi:<a href="https://doi.org/10.1016/j.jbiosc.2025.01.008">10.1016/j.jbiosc.2025.01.008</a>.</div>
</div>
<div id="ref-luecken2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">5. </div><div class="csl-right-inline">Luecken, M. D. &amp; Theis, F. J. <a href="https://doi.org/10.15252/msb.20188746">Current best practices in single<span>-</span>cell RNA<span>-</span>seq analysis: A tutorial</a>. <em>Molecular Systems Biology</em> <strong>15</strong>, e8746 (2019).</div>
</div>
<div id="ref-mangla" class="csl-entry" role="listitem">
<div class="csl-left-margin">6. </div><div class="csl-right-inline">Mangla, N., Singh, R. &amp; Agarwal, N. <a href="https://doi.org/10.1128/spectrum.00312-23">HtpG is a metal-dependent chaperone which assists the DnaK/DnaJ/GrpE chaperone system of mycobacterium tuberculosis via direct association with DnaJ2</a>. <em>Microbiology Spectrum</em> <strong>11</strong>, e00312–23.</div>
</div>
<div id="ref-choby2016" class="csl-entry" role="listitem">
<div class="csl-left-margin">7. </div><div class="csl-right-inline">Choby, J. E. &amp; Skaar, E. P. <a href="https://doi.org/10.1016/j.jmb.2016.03.018">Heme synthesis and acquisition in bacterial pathogens</a>. <em>Journal of molecular biology</em> <strong>428</strong>, 3408–3428 (2016).</div>
</div>
<div id="ref-moeck1998" class="csl-entry" role="listitem">
<div class="csl-left-margin">8. </div><div class="csl-right-inline">Moeck, G. S. &amp; Coulton, J. W. <a href="https://doi.org/10.1046/j.1365-2958.1998.00817.x">TonB-dependent iron acquisition: mechanisms of siderophore-mediated active transport</a>. <em>Molecular Microbiology</em> <strong>28</strong>, 675–681 (1998).</div>
</div>
<div id="ref-fujita2019" class="csl-entry" role="listitem">
<div class="csl-left-margin">9. </div><div class="csl-right-inline">Fujita, M. <em>et al.</em> <a href="https://doi.org/10.1038/s42003-019-0676-z">A TonB-dependent receptor constitutes the outer membrane transport system for a lignin-derived aromatic compound</a>. <em>Communications Biology</em> <strong>2</strong>, 432 (2019).</div>
</div>
<div id="ref-mezzina2016" class="csl-entry" role="listitem">
<div class="csl-left-margin">10. </div><div class="csl-right-inline">Mezzina, M. P. &amp; Pettinari, M. J. <a href="https://doi.org/10.1128/AEM.01161-16">Phasins, Multifaceted Polyhydroxyalkanoate Granule-Associated Proteins</a>. <em>Applied and Environmental Microbiology</em> <strong>82</strong>, 5060–5067 (2016).</div>
</div>
</div>
</section>
</section>
</section>
<section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes">
<hr>
<ol>
<li id="fn1"><p><strong>Chemotaxis</strong> is a biological process by which cells move in response to chemical gradients in their environment, enabling navigation toward favorable conditions and away from unfavorable ones.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="../chapters/02-Materials-and-methods.html" class="pagination-link" aria-label="Materials and Methods">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Materials and Methods</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="../chapters/04-Discussion.html" class="pagination-link" aria-label="Discussion">
        <span class="nav-page-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Discussion</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer"><div class="nav-footer-center"><div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/vgoupille/Rapport_stage/edit/main/chapters/03-Results.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li></ul></div></div></div></footer><script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","descPosition":"bottom","loop":false,"openEffect":"zoom","selector":".lightbox"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script>




</body></html>